packages/react-router/src/Match.tsx

```diff
@@ -16,7 +16,11 @@ import { renderRouteNotFound } from './renderRouteNotFound'
 import { rootRouteId } from './root'
 import type { AnyRoute } from './route'
 
-export function Match({ matchId }: { matchId: string }) {
+export const Match = React.memo(function Match({
```

Comment:
Could we name the inner `Match` as `MatchImpl`?

Same for `MatchInner` and `Outlet`.

Reply:
done